jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/newcase/NewCaseInstanceViewImpl.java

```diff
@@ -151,13 +152,24 @@ public void clearRoles() {
     }
 
     @Override
-    public void setCaseDefinitions(final List<String> definitions) {
+    public void setCaseDefinitions(final List<CaseDefinitionSummary> definitions) {
         clearCaseDefinitions();
         caseTemplatesList.setValue("");
-        for (final String definition : definitions) {
-            caseTemplatesList.addOption(definition);
-        }
+        definitions.forEach(d -> {
+            long sameNameDefinitions = definitions.stream().filter(dn -> dn.getName().equals(d.getName()) && dn.getUniqueId().equals(d.getUniqueId()) == false).count();
```

Comment:
I would personally go with _findFirst()_ instead of _count()_ and then check if the Optional returned is empty or not in the if/else part. Just because the number returned by _count()_ (when it's greater than 0) is irrelevant .